frontend/Auth/FinishResetPassword.tsx

```diff
@@ -37,30 +51,81 @@ export function FinishResetPassword(props: RouterProps) {
 
   const userEmail = tokenValidationResponse?.email;
 
+  useEffect(() => {
+    const doPasswordsMatch =
+      finishResetPasswordFormData.password ===
+      finishResetPasswordFormDataCheck.password;
+
+    if (doPasswordsMatch) {
+      setFinishResetPasswordErrors({
+        message: "",
+        passwordCheck: "",
+      });
+      setDisableSubmit(false);
+    } else {
+      setFinishResetPasswordErrors({
+        message: "",
+        passwordCheck: "Passwords do not match",
+      });
+    }
+
+    if (
+      finishResetPasswordFormData.password === "" ||
+      finishResetPasswordFormDataCheck.password === ""
+    ) {
+      setFinishResetPasswordErrors({
+        message: "",
+        passwordCheck: "",
+      });
+      setDisableSubmit(true);
+    }
+  }, [finishResetPasswordFormData, finishResetPasswordFormDataCheck]);
+
+  useEffect(() => {
+    if (passwordSaveSucceeded) {
+      props.history.push("/home");
+    }
+  }, [props.history, passwordSaveSucceeded]);
+
   const submitMutation = useMutation<
     TokenValidationResponse,
     Error,
     React.FormEvent<HTMLFormElement>
-  >((evt) => {
-    evt.preventDefault();
-    const ac = new AbortController();
-
-    if (token) {
-      return flaxFetch<TokenValidationResponse>(`/send-reset-password-email`, {
-        method: "POST",
-        signal: ac.signal,
-        body: {
-          password: finishResetPasswordFormData.password,
-          token: token,
-        },
-      });
-    } else {
-      return Promise.resolve({
-        tokenIsValid: false,
-        tokenIsExpired: false,
-      });
+  >(
+    (evt) => {
+      evt.preventDefault();
+
+      const ac = new AbortController();
```

Comment:
This abortcontroller is currently pointless because we're never calling `ac.abort()`.

To fix, we should use the abortcontroller provided by react-query. See https://react-query.tanstack.com/guides/query-cancellation. I don't know if `useMutation` passes the abortcontroller to us like `usequery`, but if so we should use that. If not, we should create a `.cancel` function on all of the promises we return from this mutation.

Assuming useMutation doesn't have an abortcontroller passed to it, here's what it would look like:

```ts
useMutation(..., () => {
  const ac = new AbortController();
  if (token) {
    const promise = flaxFetch('/api/passwords', {signal: ac.signal});
    promise.cancel = () => ac.abort()
    return promise
  }
})
```

One thing we could consider doing is automatically implementing the `.cancel` function inside of flaxFetch so that it's easier to do this in the future. Doing so would require passing the abort controller as an option to flaxFetch.

Reply:
Implemented.  Please check TypeScript solution.  Thank you!